### DISCUSSION TITLE
Handle compressed requests in API routes

### DISCUSSION THREAD

**Discussion by patrys** [vercel/next.js#61861] [Ideas]
### Goals

As it is, API routes seem to ignore `content-encoding` entirely. If you specify `content-encoding: gzip` and `content-type: application/json`, the router will call the `bodyParser` middleware, which will attempt to deserialize JSON and fail because the body is a gzipped stream.

### Background

We build apps that handle webhooks, which sometimes means hundreds of kilobytes of JSON data posted to an endpoint thousands of times per second. It would be nice to compress the data in-flight and save 80–90% of the bandwidth.

### Proposal

It would be nice to be able to add compressed payload support to API routes as easily as it's enabled for output.

I guess a workaround would be to disable the `bodyParser` middleware, and to write a custom replacement and wrap all API routes with it?

**Comment by memoday** [comment]
Any updates? Is there way to compress next api?
